src/gr11review/part2/Utility.java

```diff
@@ -1,9 +1,265 @@
+        // Var Declarations
+        int found=inner.length,count=0;
+        // Goes through the array and checks of each number exists at least
+        // once.
+        for(int i=0;i<outer.length;i++)
+            if(count<inner.length&&outer[i]==inner[count]){
+                count++; 
+                found--;
+            } 
+        if(found==0)
+            return true;
+        return false;
+    }
+
+    /**
+    * Writes the the spesified digets of pascals triangle into a file called
+    * pascalOut.txt.
+    * @author: Joshua. S 
+    */
+    public static void pascalTri(int i, int j) throws IOException{
+        // Creates buffered writer to write to file.
+        BufferedWriter bw = new BufferedWriter(new FileWriter("pascalOut.txt"));
+        // Declarations of 2D array.
+        int[][] triangle = new int[j][i];
+        // Loops through each space in the array and if it's zero on x or y its set it
+        // to 1. Then looks one space behind and one space above then adds them and
+        // sets the current one space as the sum.
+        for(int rcx=0;rcx<i;rcx++){
+            for(int ecx=0;ecx<j;ecx++){
+                if(ecx==0||rcx==0)
+                    triangle[ecx][rcx]=1;
+                else
+                    triangle[ecx][rcx] = triangle[ecx-1][rcx]+triangle[ecx][rcx-1];
+                // Writes the array value in the file
+                bw.write(""+triangle[ecx][rcx]);
+                if(ecx!=j-1)
+                    bw.write(",");
+            }
+            bw.newLine();
+        }
+        bw.close();
+    }
+  public static int sumNumbers(String str){
+    /**
+    * Given a string, return the sum of the numbers appearing in the string
+    * Ignoring all other characters
+    * @author: A. Shan
+    */
+
+    // declear the varaibles
+    int wordLength;
+    int number = 0;
+    int i;
+    int j = 0;
+    String WordNums;
+    //find the length of the word
+    wordLength = str.length();
+    // First for loop that check each character and put them in a char
+    for (i = 0; i < wordLength; i++){
+      char digit = str.charAt(i);
+      // Check if the char character is a digit or not
+      if (Character.isDigit(digit)){
+        //if it is a digit check for the digit after it
+        for (j = i + 1; j < wordLength; j++){
+          char AnotherDigit = str.charAt(j);
+          // if it is not a digit the second for loop breaks
+          if (!Character.isDigit(AnotherDigit)){
+            break;
+          }
+        }
+        // Now we use the first count and the second count as the positon of the number start and ends. We take the number as string and convert them into integer before adding them togther.
+        WordNums = str.substring(i, j);
+        number += Integer.parseInt(WordNums);
+        // When finding the next number, we need to start where it ended in the string
+        i = j;
+     }
+    }
+    //return the sum
+    return number;
+  }
```

Comment:
you don't need to comment EVERY line, only where clarification is required or general logical paragraphs/sections.  Here is a more readable take of your commenting with some comments removed.

```
    int wordLength;
    int number = 0;
    int i;
    int j = 0;
    String WordNums;
    char AnotherDigit;


    wordLength = str.length();

    for (i = 0; i < wordLength; i++){
      char digit = str.charAt(i);

       // if it is a digit check for the digit after it and replace 
      if (Character.isDigit(digit)){
        for (j = i + 1; j < wordLength; j++){
          anotherDigit = str.charAt(j);

          if (!Character.isDigit(AnotherDigit)){
            break;
          }
        }
        // use the first count and the second count as the positon of the number start and ends. We take the number as string and convert them into integer before adding them togther.
        WordNums = str.substring(i, j);
        number += Integer.parseInt(WordNums);

        //  start where it ended in the string
        i = j;
     }
    }
    //return the sum
    return number;
  }
```